pkg/storage/postgres/postgres.go

```diff
@@ -369,16 +372,25 @@ func (p *Postgres) ReadAuthorizationModels(ctx context.Context, store string, op
 		}
 	}
 
-	// TODO: make this concurrent with a maximum of 5 goroutines. This may be helpful:
-	// https://stackoverflow.com/questions/25306073/always-have-x-number-of-goroutines-running-at-any-time
-	models := make([]*openfgav1.AuthorizationModel, 0, numModelIDs)
+	g, gctx := errgroup.WithContext(ctx)
+	concurrencyControl := make(chan struct{}, p.maxConcurrency)
+	models := make([]*openfgav1.AuthorizationModel, numModelIDs)
 	// We use numModelIDs here to avoid retrieving possibly one extra model.
 	for i := 0; i < numModelIDs; i++ {
-		model, err := p.ReadAuthorizationModel(ctx, store, modelIDs[i])
-		if err != nil {
-			return nil, nil, err
-		}
-		models = append(models, model)
+		concurrencyControl <- struct{}{}
+		i := i // This line is not needed if using Go 1.22+.
+		g.Go(func() error {
+			defer func() { <-concurrencyControl }()
+			model, err := p.ReadAuthorizationModel(gctx, store, modelIDs[i])
```

Comment:
Care should be taken not to cancel the context provided to `ReadAuthorizationModel`. Doing so churns connections and causes perf regressions as a side-effect of doing so.